core/mavlink_commands.cpp

```diff
@@ -149,55 +147,51 @@ void MAVLinkCommands::receive_command_ack(mavlink_message_t message)
         case MAV_RESULT_ACCEPTED:
             _state = State::NONE;
             _state_mutex.unlock();
-            call_callback(work->callback, Result::SUCCESS, 1.0f);
-            _work_queue.pop_front();
             _parent.unregister_timeout_handler(_timeout_cookie);
+            _work_queue.pop_front();
```

Comment:
What makes you think that those two calls should be inverted? Just to be sure I understand it :-).

Reply:
See commit message but basically you could have popped the work item from the queue and then the timeout still triggers and wants to access this work item.